#!/usr/bin/env python
PACKAGE = "hfts_grasp_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('min_iterations', int_t, 0, "Minimum number iterations", 20, 1, 100000)
gen.add('max_iterations', int_t, 0, "Maximum number iterations", 70, 1, 100000)
gen.add('free_space_weight', double_t, 0, "Weight for free space proximity", 0.5, 0.0, 10.0)
gen.add('connected_space_weight', double_t, 0, "Weight for connected space proximity", 4.0, 0.0, 10.0)
gen.add('max_num_hierarchy_descends', int_t, 0, "Maximum number of hierarchy descends per goal sample call", 0, 0, 20)
gen.add('com_center_weight', double_t, 0, "Weight for grasp quality measure for Robotiq hand.", 0.1, 0.0, 10.0)
gen.add('reachabilty_weight', double_t, 0, "Weight for reachability component in objective function", 1.0, 0.0, 1000.0)
gen.add('time_limit', double_t, 0, "Maximal planning time is seconds", 60.0, 0.0, 6000.0)
gen.add('use_approximates', bool_t, 0, "Flag whether to use approximate goals", True)
gen.add('compute_velocities', bool_t, 0, "Flag whether to compute velocities", True)
gen.add('max_normal_variance', double_t, 0, "Maximal allowed variance of normal in the neighborhood of a contact point",
        0.2, 0.0, 100.0)
gen.add('min_contact_patch_radius', double_t, 0, "Size of the contact patch for which the normal variance is limitied",
        0.01, 0.0, 1.0)
gen.add('max_num_points', int_t, 0, "Maximal number of contact points for an object", 10000, 100, 100000)
gen.add('contact_density', double_t, 0, "Number of contact points per m^2", 300.0, 1.0, 1000.0)
gen.add('hfts_position_weight', double_t, 0, "Weight for positions in HFTS clustering", 1.0, 0.0, 100.0)
gen.add('velocity_factor', double_t, 0, "Scaling factor for maximal hand/arm velocity", 0.2, 0.0, 1.0)
gen.add('hfts_branching_factor', int_t, 0, "HFTS branching factor", 4, 2, 100)
gen.add('hfts_first_level_branching_factor', int_t, 0, "HFTS branching factor on first level", 3, 2, 100)
gen.add('force_new_hfts', bool_t, 0, "Flag whether to force recomputing the hfts", False)

exit(gen.generate(PACKAGE, "hfts_integrated_planner_node", "IntegratedHFTSPlanner"))